inst/templates/monolix-step-01-preprocess.Rmd

```diff
@@ -0,0 +1,180 @@
+pk_data %>% 
+  distinct(group, analyte, route, dose, dose_units) %>% 
+  arrange(group, analyte) %>% 
+  kable() %>% 
+  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
+```
+
+```{r check-units}
+
+# check other units that are used for calculating covariate values
+
+pk_data %>% distinct(weight_unit, height_unit)
+```
+
+<!-- Usually, we evaluate participant weight, BMI, age, sex as covariates in the PK model. Baseline serum albumin is another covariate worth investigating. Bajaj et. al (2019) found that baseline body weight, serum albumin, and sex are the most commonly identified covariates in oncology PK models (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6710515/).  -->
+
+<!-- At pre-infusion time points, concentrations of the analyte should be LLoQ. If not, plot these data and have a discussion about how you will take it into account in the PK model. -->
+
+```{r check-pre-infusion}
+
+# check for concentrations > LLOQ at pre-infusion time points
+pk_data %>% 
+  filter(visit_type == "1.00 A") %>% 
+  filter(concentration > lloq) %>% 
+  select(pubid, arm, group, concentration, concentration_units) %>% 
+  kable() %>% 
+  kable_styling(bootstrap_options = c("striped", "hover", "condensed", "responsive"))
+```
+
+## Format data for Monolix/PKanalix
+
+<!-- 
+
+The next step is to create the CSV files for use with Monolix or PKanalix. 
+
+Data specs for these CSV files: https://github.com/FredHutch/VISC-Assay-Methods/blob/master/PK/Assay_Data_Specs/Monolix_data.md 
+
+-->
+
+<!-- You will need to filter out the placebo participants. You can do this by using the unblinding variable if it is available. If you don't have the unblinding variable, you can filter out participants who have all concentrations <= LLOQ. -->
+
+```{r pre-process-data}
+
+# calculate any variables that you need and are not in the data package
+cvdxxx_elisa_preprocess <- pk_data %>%
+  filter(analyte != "Placebo") %>% 
+  mutate(
+    dose_amount = dose*weight,
+    bmi = weight/((height/100)**2),
+    censored = if_else(concentration_ug <= lloq_ug, 1, 0),
+    ) %>% 
+  select(
+    pubid, analyte, group, hiv_status, route, dose, dose_units, dose_amount,
+    infusion_no, infusion_dur,
+    visitno, day, study_day, concentration, lloq, censored,
+    sex, age, weight, height, bmi
+    ) 
+
+# export data 
+usethis::use_data(cvdxxx_elisa_preprocess, overwrite = TRUE)
+```
+
+<!-- Now create the datasets that will be exported to CSVs for Monolix. If you don't have a row for the infusion time point in the data (day == 0), you will need to add that in. -->
+
+```{r monolix-data}
+
+# Monolix uses "." to designate missing values
+# For each participant, the first row should be day == 0 with the dose amount (AMT). The concentration (DV) is "." at this time point.
+
+monolix_data <- cvdxxx_elisa_preprocess %>% 
+  filter(day >= 0) %>% # we don't use pre-infusion time points in the PK analysis
+  group_by(pubid, infusion_no) %>% 
```

Comment:
`group_by(analyte, pubID, infusion_no` # in many studies we are receiving more than one analyte